packages/sourcecred/src/cli/update/utils.js

```diff
@@ -0,0 +1,24 @@
+// @flow
+
+import cloneDeep from "lodash.clonedeep";
+
+// This is a type-friendly way to add optional attributes to a config.
+// Attributes from the `optionals` parameter will only be added if they
+// are not undefined.
+// It is recommended to define T explicitly when using this.
+// Example: `const object = buildObject<MyType>(...);`
+export function buildObject<T: {}>(
+  required: T,
+  optionals: $Shape<T>,
+  exclusions: $ReadOnlyArray<$Values<T> | typeof undefined> = [undefined]
```

Comment:
What's wrong with this?
```suggestion
  exclusions: $ReadOnlyArray<$Values<T>> = []
```
Is it some idiosyncrasy in the flow parser, or are you actually expecting some array entries to be undefined?

Reply:
Actually expecting undefined values in `optionals` as the `exclusions` are the values that should be not be passed from `optionals` into the result.